wpiutil/src/main/java/edu/wpi/first/util/sendable2/SendableTable.java

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.util.sendable2;
+
+import edu.wpi.first.util.function.BooleanConsumer;
+import edu.wpi.first.util.function.FloatConsumer;
+import edu.wpi.first.util.function.FloatSupplier;
+import edu.wpi.first.util.protobuf.Protobuf;
+import edu.wpi.first.util.struct.Struct;
+import edu.wpi.first.util.struct.StructBuffer;
+
+import java.nio.ByteBuffer;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.DoubleConsumer;
+import java.util.function.DoubleSupplier;
+import java.util.function.LongConsumer;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+
+/** Helper class for building Sendable dashboard representations. */
+public interface SendableTable extends AutoCloseable {
+  void setBoolean(String name, boolean value);
+
+  void publishBoolean(String name, BooleanSupplier supplier);
+
+  BooleanConsumer publishBoolean(String name);
+
+  void subscribeBoolean(String name, BooleanConsumer consumer);
+
+  void setInt(String name, long value);
+
+  void publishInt(String name, LongSupplier supplier);
+
+  LongConsumer publishInt(String name);
+
+  void subscribeInt(String name, LongConsumer consumer);
+
+  void setFloat(String name, float value);
+
+  void publishFloat(String name, FloatSupplier supplier);
+
+  FloatConsumer publishFloat(String name);
+
+  void subscribeFloat(String name, FloatConsumer consumer);
+
+  void setDouble(String name, double value);
+
+  void publishDouble(String name, DoubleSupplier supplier);
+
+  DoubleConsumer publishDouble(String name);
+
+  void subscribeDouble(String name, DoubleConsumer consumer);
+
+  void setString(String name, String value);
+
+  void publishString(String name, Supplier<String> supplier);
+
+  Consumer<String> publishString(String name);
+
+  void subscribeString(String name, Consumer<String> consumer);
+
+  void setRaw(String name, String typeString, byte[] value, int start, int len);
+
+  void setRaw(String name, String typeString, ByteBuffer value, int start, int len);
+
+  void publishRaw(String name, String typeString, Supplier<byte[]> supplier);
+
+  void publishRawBB(String name, String typeString, Supplier<ByteBuffer> supplier);
+
+  Consumer<byte[]> publishRaw(String name, String typeString);
+
+  Consumer<ByteBuffer> publishRawBB(String name, String typeString);
```

Comment:
I'm not a fan of the "Raw"/"RawBB" naming distinction; it's not very intuitive. How about "RawBytes"/"RawBuffer" respectively?

Reply:
Also, would it be possible to implement the struct methods using `byte[]` instead of `ByteBuffer`? `ByteBuffer.get(byte[] arr)` would be doable, though for efficiency the destination `byte[]` would require caching somewhere. (Might be doable to store the `byte[]` in `StructBuffer`, but I don't know if that would make sure)

Reply:
The problem is the actual Struct read/write implementations need a ByteBuffer object to e.g. write floats or int values.  Using ByteBuffer also enables avoiding copies within the native code interfaces.  I agree in general I think the ByteBuffer API is less intuitive with the whole position/limit thing, but we also can't get rid of it.